File: backend/routes/announcement.routes.js
import { Router } from 'express';
import {
  getAnnouncements,
  createAnnouncement,
  deleteAnnouncement,
} from '../controllers/announcement.controller.js';
import { verifyToken, requireRole } from '../middleware/auth.middleware.js';

const router = Router();

router.get('/', getAnnouncements);
router.post('/', verifyToken, requireRole('admin', 'teacher'), createAnnouncement);
router.delete('/:id', verifyToken, requireRole('admin'), deleteAnnouncement);

export default router;

File: backend/routes/announcement_routes.py
from flask import Blueprint
from controllers.announcement_controller import (
    get_announcements,
    create_announcement,
    delete_announcement,
)
from middleware.auth_middleware import verify_token, require_role

announcement_bp = Blueprint('announcements', __name__)

announcement_bp.route('/', methods=['GET'])(get_announcements)
announcement_bp.route('/', methods=['POST'])(
    verify_token(require_role('admin', 'teacher')(create_announcement))
)
announcement_bp.route('/<string:announcement_id>', methods=['DELETE'])(
    verify_token(require_role('admin')(delete_announcement))
)
